src/profile/achievements.ts

```diff
@@ -8,8 +7,7 @@ export class Achievements extends PageParser {
         return achievements;
     }
 
-    protected getURL(req: Request): string {
-        return "https://na.finalfantasyxiv.com/lodestone/character/" + req.params.characterId + "/achievement";
+    protected getLodestonePage(characterId: string): Promise<string> {
```

Comment:
The original purpose of passing the `Request` object was to give the class access to query string and params at the same time, since query string could also contain paging information for when it'll be implemented, what are your plans regarding that?

Reply:
I think we should use a similar approach like the options object parameter in the `get` method here.

I would strongly advise against using a type from an unrelated library (`Request` is from express here) as users of nodestone might not actually use express in their code (e.g. for Discord bots or just random local scripts).

I don't think there is any better way than just extending the options object with everything needed for a parser (e.g. paging information) and then passing it through to the `getURL`/`getLodestonePage` function, which might sound like a lot of work. But with the current `Request` solution you would want to document all needed parameters of the `Request` object anyway, so people who don't use express can mock it without running into errors, which is basically the same amount of work. Then this would just eliminate the `Request` type from the nodestone lib and make the interface more explicit, since you have the expected parameters as types/generated docs rather than manually maintained docs.

Reply:
Yeah, I strongly agree with that, moving to our own Option object will greatly help with that :)